src/modelframe.jl

```diff
@@ -172,19 +172,40 @@ function setcontrasts!(mf::ModelFrame, new_contrasts::Dict)
 end
 setcontrasts!(mf::ModelFrame; kwargs...) = setcontrasts!(mf, Dict(kwargs))
 
+convertinsupport(v::AbstractVector, d::Distribution) = Vector{partype(d)}(v)
+
+function convertinsupport(v::Union{AbstractCategoricalVector, AbstractNullableCategoricalVector},
+    d::Bernoulli)
+    vdrp = droplevels!(copy(v))
+    levs = CategoricalArrays.levels(vdrp)
+    llv = length(levs)
+    if llv < 2
+        zeros(vdrp, partype(d))
+    elseif llv == 2
+        partype(d)[get(x) == levs[2] for x in vdrp]
+    else
+        throw(ArgumentError("length(levels(droplevels!(v))) = $llv, should be ≤ 2"))
+    end
+end
+
+function convertinsupport(v::Vector{String}, d::Bernoulli)
```

Comment:
You'll need a method for `AbstractVector{Nullable{String}}` too. And this one should be made more general, accepting any `AbstractVector{String}`.